### PR TITLE
Python 3.11

### PR DESCRIPTION
**Pull request by satmandu**
- Python 3.11
- Steps used to update to python3.11:
  - Update `python3` package to `3.11.0` with `ensurepip` enabled
  - Then...

```
crew upgrade python3 
pip3 install --upgrade pip 
pip3 install build setuptools pyproject-hooks wheel build installer tomli
```
  - Build (and update as necessary) the following packages in this order:
  - ` py3_setuptools.rb py3_packaging.rb py3_wheel.rb py3_flit_core.rb py3_tomli.rb py3_pyproject_hooks.rb py3_build.rb py3_installer.rb py3_pyparsing.rb py3_pip.rb py3_pep517.rb py3_libxml2.rb py3_pyelftools.rb py3_codespell.rb`
  - Add binaries to those packages, and put them into this PR, and then reopen a fresh container and update to that PR and do a `crew upgrade`
  - Rebuild everything else.
  
Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=python311 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/python3.rb [hunk 1]
```diff
@@ -70,12 +72,12 @@ def self.build
...
     # So, modifying environment variable to make pass tests.
 
-    Dir.mkdir 'builddir'
+    FileUtils.mkdir_p 'builddir'
     Dir.chdir 'builddir' do
       system CREW_ENV_OPTIONS_HASH.transform_values { |v| "#{v} #{@cppflags}" }, "../configure #{CREW_OPTIONS} \
         --with-computed-gotos \
         --enable-loadable-sqlite-extensions \
-        --without-ensurepip \
+        --with-ensurepip \
```
Comment (uberhacker) [inline packages/python3.rb:80]: I agree.  We should be bundling pip with python.  That said, I think python needs to be removed prior to building so it integrates correctly.  Does this mean the `py3_pip` package goes away?

Comment (satmandu) [reply packages/python3.rb:80]: That would be the idea. It would be nice to just make crew a frontend for pip3 for python3 packages...

**Comment by satmandu** [comment]
All builds except these succeeded:
```
py3_cmarkgfm.rb
py3_cryptography.rb
py3_cython.rb
py3_greenlet.rb
py3_lxml.rb
py3_numpy.rb
```

**Comment by satmandu** [comment]
Ok just waiting on the completion of the armv7l builds of py3_lxml.rb, py3_numpy.rb, and meson.rb.

**Comment by uberhacker** [comment]
> All builds except these succeeded:
> 
> ```
> py3_cmarkgfm.rb
> py3_cryptography.rb
> py3_cython.rb
> py3_greenlet.rb
> py3_lxml.rb
> py3_numpy.rb
> ```

Is this still the case?  Have you tried to look for new releases?

**Comment by satmandu** [comment]
> > All builds except these succeeded:
> > 
> > ```
> > py3_cmarkgfm.rb
> > py3_cryptography.rb
> > py3_cython.rb
> > py3_greenlet.rb
> > py3_lxml.rb
> > py3_numpy.rb
> > ```
> 
> Is this still the case?  Have you tried to look for new releases?

Those have all been successfully updated.

File: packages/py3_gi_docgen.rb [hunk 2]
```diff
@@ -19,10 +19,10 @@ class Py3_gi_docgen < Package
...
   })
   binary_sha256({
-    aarch64: '36720faa65b83944a74384038f50d878ae21b8f2cc076fae7feb3c065216ad91',
-     armv7l: '36720faa65b83944a74384038f50d878ae21b8f2cc076fae7feb3c065216ad91',
-       i686: '87267762d0360ac2f65c21a3a38efd186ccbe8d3f2869bdf9c0d64924fd40630',
-     x86_64: '61da242a83085c27b3f1a7b4e4966fa28568f4654f430e638a7764e38111ba56'
+    aarch64: '65c999d122794719e33d424824c74c5652e096cfc00602c3d4c3735580ed890c',
+     armv7l: '65c999d122794719e33d424824c74c5652e096cfc00602c3d4c3735580ed890c',
+       i686: '7fb80e8f5ceb7f4203648bf8e31cdf1447ff2e9b7a705b8cb915b21988d4ac26',
+     x86_64: 'b032122aca3708e986ea8ff5b1b6a978407ee9c7370006c560e3c865b4242cfd'
```
Comment (uberhacker) [inline packages/py3_gi_docgen.rb:25]: Version missing `-py3.11`?

Comment (satmandu) [reply packages/py3_gi_docgen.rb:25]: Rebuilding...

File: packages/py3_pyproject_hooks.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    aarch64: '21f517d2022987ada64860830426cb7f7a769e405d42cb381f24063629433e55',
+     armv7l: '21f517d2022987ada64860830426cb7f7a769e405d42cb381f24063629433e55',
+       i686: '3de0b5be888db9484499d5083ed1011908945e36603fff4a2386869a5ba897ce',
+     x86_64: 'bc0168df62eb6b2c238a49bbfa289cdb1e3941648e8d6427552a9270fefc6579'
+  })
+
+  depends_on 'python3'
+  depends_on 'py3_tomli'
+
+  conflicts_ok
```
Comment (uberhacker) [inline packages/py3_pyproject_hooks.rb:29]: What are the conflicts?

File: packages/py3_gi_docgen.rb (hunk 2 above)
Comment (satmandu) [reply packages/py3_gi_docgen.rb]: Fixed

File: packages/py3_pyproject_hooks.rb (hunk 3 above)
Comment (satmandu) [reply packages/py3_pyproject_hooks.rb:29]: Conflicts are only for the first tranche of packages built after first installing the new version of python...

**Review by uberhacker — APPROVED** [review]

**Pull request merged**